proposed/2021/DotnetNugetTrustedSigners.md

```diff
@@ -0,0 +1,196 @@
+# Title
+
+- Author Name [Kat Marchan (@zkat)](https://github.com/zkat)
+- Start Date 2021-03-04
+- GitHub Issue (GitHub Issue link)
+- GitHub PR (GitHub PR link)
+
+## Summary
+
+This specifies the syntax and behavior for the `dotnet.exe nuget trusted-signers` command.
+
+## Motivation
+
+This is part of a larger effort to reach full parity between `nuget.exe` and
+`dotnet.exe` for NuGet-related operations.
+
+## Explanation
+
+### Functional explanation
+
+#### `dotnet nuget trusted-signers`
+
+```
+Provides the ability to manage the list of trusted signers.
+
+USAGE:
+    dotnet nuget trusted-signer [OPTIONS] [COMMAND]
+
+COMMANDS:
+    list (default)
+    trust-package <name> <package>...
+    trust-source <name> [source-url]
+    trust-certificate <name> <fingerprint>
+    remove <name>
+    sync <name>
+```
+
+#### `dotnet nuget trusted-signers list`
+
+```
+Lists all the trusted signers in the configuration. This option will include all the certificates (with fingerprint and fingerprint algorithm) each signer has. If a certificate has a preceding [U], it means that certificate entry has allowUntrustedRoot set as true.
+
+USAGE:
+    dotnet nuget trusted-signers [OPTIONS] list
+
+OPTIONS:
+    -h, --help                          Prints help information.
+    -v, --verbosity <LEVEL>             Set the MSBuild verbosity level. Allowed values are q[uiet], m[inimal], n[ormal], d[etailed], and diag[nostic].
+        --interactive                   Allows command to stop and wait for user input or action.
+        --configfile                    The NuGet configuration file. If specified, only the settings from this file will be used. If not specified, the hierarchy of configuration files from the current directory will be used.
+
+EXAMPLE:
+    $ dotnet nuget trusted-signers list
+    Registered trusted signers:
+
+    1.   nuget.org [repository]
+        Service Index: https://api.nuget.org/v3/index.json
+        Certificate fingerprint(s):
+            SHA256 - 0E5F38F57DC1BCC806D8494F4F90FBCEDD988B46760709CBEEC6F4219AA6157D
+
+    2.   microsoft [author]
+        Certificate fingerprint(s):
+            SHA256 - 3F9001EA83C560D712C24CF213C3D312CB3BFF51EE89435D3430BD06B5D0EECE
+            SHA256 - AA12DA22A49BCE7D5C1AE64CC1F3D892F150DA76140F210ABD2CBFFCA2C18A27
+
+    3.   myUntrustedAuthorSignature [author]
+        Certificate fingerprint(s):
+            [U] SHA256 - 518F9CF082C0872025EFB2587B6A6AB198208F63EA58DD54D2B9FF6735CA4434
+```
+
+#### `dotnet nuget trusted-signers sync`
```

Comment:
@chgill-MSFT what's the concern with `nuget.exe sync` again? I remember we chose not to advertise this command on this blog post: https://devblogs.microsoft.com/nuget/the-nuget-org-repository-signing-certificate-will-be-updated-as-soon-as-march-15th-2021/

Is there something we should do differently for dotnet.exe's command?

Reply:
> Note: This gesture will delete the current list of certificates and replace them with an up-to-date list from the repository.

Maybe this? We should probably still implement it.